### PR TITLE
Feat/Expose spin counter

### PR DESCRIPTION
**Pull request by lhruby**
# Description
Expose spin counter in `Spinner` class. This e.g. allows to use the counter to assign monotonous ids to data processed during `spinOnce`. A typical example is a `frame_id` or `frame_index`. 

## Type of change
- New feature (non-breaking change which adds functionality)

## Checklist before requesting a review
- [x] I have performed a self-review of my code.
- [ ] If it is a core feature, I have added thorough tests.
- [ ] If this is a new component I have added examples.
- [ ] I updated the README and related documentation.

### CODE REVIEW

**Review by filippobrizzi — APPROVED** [review]

File: modules/concurrency/src/spinner.cpp [hunk 1]
```diff
@@ -89,4 +89,8 @@ void Spinner::addStopCallback(std::function<void()>&& callback) {
   stop_callback_ = std::move(callback);
 }
 
+auto Spinner::spinCount() const -> uint64_t {
+  return spin_count_;
```
Comment (floriantschopp) [inline modules/concurrency/src/spinner.cpp:93]: its never incremented and the user has no idea that he needs to in `spinOnce`... Should we wrap spinOnce into a `spinOnceParent` (very bad name) sth like
```
void spinOnceParent() {
   ++spin_count_;
   spinOnce();
}
```

**Review by floriantschopp — CHANGES_REQUESTED** [review]
somewhat unrelated but still a bit related: 
Most of the time, I implenent spinOnce to just sleep. Instead of being pure virtual, should we give a default implementation that just sleeps?

File: modules/concurrency/src/spinner.cpp (hunk 1 above)
Comment (floriantschopp) [reply modules/concurrency/src/spinner.cpp:93]: at the very least, we need to leave a comment

Comment (filippobrizzi) [reply modules/concurrency/src/spinner.cpp:93]: it is incremented in `spin()` line 64

Comment (lhruby) [reply modules/concurrency/src/spinner.cpp:93]: You actually increment it in `spin()`
`    const auto target_timestamp = start_timestamp_ + (++spin_count_) * spin_period_;`

**Comment by lhruby** [comment]
> somewhat unrelated but still a bit related: Most of the time, I implenent spinOnce to just sleep. Instead of being pure virtual, should we give a default implementation that just sleeps?

what does a spinner do which simply sleeps? can you give an example?

**Comment by filippobrizzi** [comment]
> somewhat unrelated but still a bit related: Most of the time, I implenent spinOnce to just sleep. Instead of being pure virtual, should we give a default implementation that just sleeps?

@floriantschopp  What is the use case for having a spinner that just sleep?

Also, I don't know if you saw my message on discord. 
But we should update this design. The proper way would be instead of using inheritance is to use composition and have `Spinner` get `spinOnce` as callback.
If I do this refactor is going to be a problem for you? do you already use Spinner in many places?

File: modules/concurrency/src/spinner.cpp (hunk 1 above)
Comment (floriantschopp) [reply modules/concurrency/src/spinner.cpp:93]: sorry, missed that

**Comment by floriantschopp** [comment]
we can change the interface, thats ok for now

**Pull request merged**